src/web/src/components/Posts/GitHubInfo.tsx

```diff
@@ -25,37 +26,64 @@ type Props = {
   ghUrls: string[];
 };
 
-const filterGitHubUrls = (ghUrls: string[]) => {
+const filterGitHubUrls = (urls: string[]) => {
   const issues: string[] = [];
   const pullRequests: string[] = [];
   const repos: string[] = [];
+  const commits: string[] = [];
 
-  ghUrls.forEach((ghUrl) => {
-    // add unique repo urls to repos array
-    if (
-      ghUrl.match(/https:\/\/github\.com\/[^\/]+\/[^\/]+/) &&
-      !repos.includes(ghUrl.replace(/(https:\/\/github\.com\/[^\/]+\/[^\/]+)\/.*/, '$1'))
-    ) {
-      repos.push(ghUrl.replace(/(https:\/\/github\.com\/[^\/]+\/[^\/]+)\/.*/, '$1'));
-    }
+  const ghUrls = urls.map(parseGitHubUrl).filter((url) => url !== null) as URL[];
+
+  for (const url of ghUrls) {
+    const { pathname, href } = url;
 
-    // remove trailing characters after issue or pullRequest number from ghUrl
-    const trimmedUrl: string = ghUrl.replace(
-      /(https:\/\/github\.com\/[^\/]+\/[^\/]+\/(issues|pull)\/[0-9]+).*/,
-      '$1'
+    // Match urls that start with /<user>/<repo>, and optionally end with /<anything-in-between>/<type>/<id>
+    // Ex: /Seneca-CDOT/telescope/pull/2367 ✅
+    // Ex: /Seneca-CDOT/telescope ✅
+    // Ex: /Seneca-CDOT/telescope/pull/2367/commits/d3fag ✅
+    // Ex: /Seneca-CDOT/telescope/issues ✅
+    const matches = /^\/(?<user>[^\/]+)\/(?<repo>[^\/]+)((\/(.*))?\/(?<type>[^\/]+)\/(?<id>(\w+))$)?/i.exec(
+      pathname
     );
+    if (matches?.groups === undefined) continue;
+    const { type, user, repo } = matches.groups;
+
+    const repoUrl = `https://github.com/${user}/${repo}`;
+    !repos.includes(repoUrl) && repos.push(repoUrl);
```

Comment:
Let's use an `if` here please.

Reply:
Question: why not use a `Set()` instead of an Array to build all these, so you can ignore all the `includes?` checks.  Then you can convert them to `Array` before you return.